src/api/procedures/addConfidentialTransaction.ts

```diff
@@ -0,0 +1,320 @@
+import { u64 } from '@polkadot/types';
+import {
+  PalletConfidentialAssetTransactionLeg,
+  PolymeshPrimitivesMemo,
+} from '@polkadot/types/lookup';
+import { ISubmittableResult } from '@polkadot/types/types';
+import BigNumber from 'bignumber.js';
+
+import {
+  assertConfidentialAccountExists,
+  assertConfidentialAssetExists,
+  assertConfidentialAssetsEnabledForVenue,
+  assertConfidentialVenueExists,
+  assertIdentityExists,
+} from '~/api/procedures/utils';
+import {
+  ConfidentialTransaction,
+  Context,
+  DefaultPortfolio,
+  NumberedPortfolio,
+  PolymeshError,
+  Procedure,
+} from '~/internal';
+import {
+  AddConfidentialTransactionParams,
+  AddConfidentialTransactionsParams,
+  ConfidentialAssetTx,
+  ErrorCode,
+  RoleType,
+} from '~/types';
+import { BatchTransactionSpec, ProcedureAuthorization } from '~/types/internal';
+import { MAX_LEGS_LENGTH } from '~/utils/constants';
+import {
+  auditorsToBtreeSet,
+  bigNumberToU64,
+  confidentialAccountToMeshPublicKey,
+  confidentialAssetsToBtreeSet,
+  confidentialLegToMeshLeg,
+  confidentialTransactionIdToBigNumber,
+  identitiesToBtreeSet,
+  stringToMemo,
+} from '~/utils/conversion';
+import {
+  asConfidentialAccount,
+  asConfidentialAsset,
+  asIdentity,
+  assembleBatchTransactions,
+  filterEventRecords,
+  optionize,
+} from '~/utils/internal';
+
+/**
+ * @hidden
+ */
+export type Params = AddConfidentialTransactionsParams & {
+  venueId: BigNumber;
+};
+
+/**
+ * @hidden
+ */
+export interface Storage {
+  portfoliosToAffirm: (DefaultPortfolio | NumberedPortfolio)[][];
+}
+
+/**
+ * @hidden
+ */
+type InternalAddTransactionParams = [
+  u64, // venueID
+  PalletConfidentialAssetTransactionLeg[],
+  PolymeshPrimitivesMemo | null
+][];
+
+/**
+ * @hidden
+ */
+export const createConfidentialTransactionResolver =
+  (context: Context) =>
+  (receipt: ISubmittableResult): ConfidentialTransaction[] => {
+    const events = filterEventRecords(receipt, 'confidentialAsset', 'TransactionCreated');
+
+    const result = events.map(
+      ({ data }) =>
+        new ConfidentialTransaction({ id: confidentialTransactionIdToBigNumber(data[2]) }, context)
```

Comment:
@prashantasdeveloper FYI, I had a bug here on the original PR. I did `data[1]`, which is the venue ID.